bigquery/google/cloud/bigquery/client.py

```diff
@@ -1439,11 +1439,12 @@ def _check_mode(stream):
     :param stream: A bytes IO object open for reading.
 
     :raises: :exc:`ValueError` if the ``stream.mode`` is a valid attribute
-             and is not among ``rb``, ``r+b`` or ``rb+``.
+             and is not among ``rb``, ``r+b``, ``rb+``, or, in case of gzip,
+             not among ``1`` or ``2``.
     """
     mode = getattr(stream, 'mode', None)
 
```

Comment:
Rather than mixing in the oddball, let's add an `if isinstance(stream, gzip.GzipFile)` test here, e.g.:

```python
import gzip
...

        if isinstance(stream, gzip.GzipFile):
            readable = mode == gzip.READ
        else:
            readable = mode in ('rb', 'r+b', 'rb+')
        if not readable:
            raise ValueError(...)